registry/storage/driver/azure/azure.go

```diff
@@ -506,26 +486,16 @@ func (w *writer) Commit() error {
 }
 
 type blockWriter struct {
-	client    azure.BlobStorageClient
-	container string
-	path      string
+	ctx    context.Context
```

Comment:
"Do not store contexts in a struct type" is one of the fundamental rules of contexts. You have a justified reason to break that rule here (needing to satisfy `io.Writer`) but please document the reasoning in a code comment.

Reply:
makes sense, added a comment